### ISSUE TITLE
eemumu_AV depends on CUDA for Host Only CPU when CUDA is not defined

### ISSUE THREAD

**Issue by lfield** [madgraph5/madgraph4gpu#38]
Even when the __CUDACC__ is not defined, eemumu_AV depends on CUDA for Host Only CPU

**Comment by roiser** [comment]
Despite this would also work in a e.g. CPU only workflow (the random numbers of curand can be generated on the host and stay on the host) it would be better not to depend on cuda if not necessary. As this is one of our few external dependencies it could make sense to introduce another switch on which random number generator we want to depend. 

For completeness @valassi  found a RN generator for AMD / RocM which seems to be spawned from the cuda one at 

https://github.com/ROCmSoftwarePlatform/rocRAND

**Comment by lfield** [comment]
It works on the CPU if I install the CUDA libraries. Is there a concern that if we use different RN generators, on different devices, it could lead to different results?

**Comment by roiser** [comment]
My believe (others pls correct me) is that we have to face this situation anyway once we go into production and we have to verify our workflow correctness with other means than comparing individual results. 

Having said this for development it may be a nice feature to run the same RN generator both on the host and the device (as e.g. its possible for curand). As curand would produce the same RN starting from the same seed on host and device this can come handy for debugging.

**Comment by lfield** [comment]
It looks like SYCL does not play nicely with CUDA (redefinitions).

**Comment by hageboeck** [comment]
I have it compiling in one of my branches without cuda. Will put up a PR soon-ish.
Caveat:
Need to move from curand to c++ random using
`make MGONGPU_CONFIG=-DMGONGPU_COMMONRAND_ONHOST`
or
`export MGONGPU_CONFIG=-DMGONGPU_COMMONRAND_ONHOST`

That could be made more accessible, but I didn't want to interfere with the "cuda is the default" config.

**Comment by valassi** [comment]
Hi @hageboeck thanks. I know the hardcoded mgOnGpuConfig.h is a nuisance now. Note that the -D flags will not work, because you also have to undefine the hardcoded other defaults. If Laurence only needs this in his branch/directory for sycl (which is ONLY used for sycl), I would suggest that he simply changes the mgOnGpuConfig.h to have
```
// Curand random number generation (CHOOSE ONLY ONE)
//#define MGONGPU_CURAND_ONDEVICE 1 // default (curand: CUDA on device, C++ on host)
//#define MGONGPU_CURAND_ONHOST 1 // (curand: CUDA on host, C++ on host)
#define MGONGPU_COMMONRAND_ONHOST 1 // (common rand: CUDA on host, C++ on host)
```
instead of
```
// Curand random number generation (CHOOSE ONLY ONE)
#define MGONGPU_CURAND_ONDEVICE 1 // default (curand: CUDA on device, C++ on host)
//#define MGONGPU_CURAND_ONHOST 1 // (curand: CUDA on host, C++ on host)
//#define MGONGPU_COMMONRAND_ONHOST 1 // (common rand: CUDA on host, C++ on host)
```
Cheers
Andrea

**Comment by hageboeck** [comment]
They will work. See PR that's coming up.

**Comment by hageboeck** [comment]
Fixed in #59
